marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/extensions/runFlow.sjs

```diff
@@ -16,30 +16,41 @@
 'use strict';
 const DataHub = require("/data-hub/5/datahub.sjs");
 
-//todo flush this out
 function get(context, params) {
   return post(context, params, null);
 }
 
 function post(context, params, input) {
   let flowName = params["flow-name"];
-
+  let stepNumber = params.step;
   if (!fn.exists(flowName)) {
     fn.error(null, "RESTAPI-SRVEXERR", Sequence.from([400, "Bad Request", "Invalid request - must specify a flowName"]));
   } else {
     let options = params["options"] ? JSON.parse(params["options"]) : {};
     const datahub = new DataHub({ performanceMetrics: !!options.performanceMetrics });
     let jobId = params["job-id"];
     let flow = datahub.flow.getFlow(flowName);
-    let targetDatabase = params["target-database"] ? xdmp.database(params["target-database"]) : xdmp.database(datahub.config.FINALDATABASE);
-
+    let stepRef = flow.steps[stepNumber];
+    let stepDetails = datahub.flow.step.getStepByNameAndType(stepRef.stepDefinitionName, stepRef.stepDefinitionType);
+    let flowOptions = flow.options || {};
+    let stepRefOptions = stepRef.options || {};
+    let stepDetailsOptions = stepDetails.options || {};
+    // build combined options
+    let combinedOptions = Object.assign({}, flowOptions, stepDetailsOptions, stepRefOptions, options);
+    // combine all collections
```

Comment:
I see differences with the 'options' precedence in different places.

collector.sjs -> let combinedOptions = Object.assign({}, baseStep.options, stepDoc.options, flowDoc.options, options);  - 'options' 's precedence is different. I am using this order for WriteStepRunner on the client side

impl/flow.sjs ->    let combinedOptions = Object.assign({}, stepDetails.options, stepRef.options, options); - flow options not used

Reply:
Good catch. We should definitely be consistent. The way I see it the precedence should be 1. options passed, 2. stepRef/baseStep (inside flow), 3. flow options, and 4. stepDetails/stepDoc.

@aebadirad, is that the way you see it?

Reply:
yes sir.